homeassistant/components/nest/media_source.py

```diff
@@ -56,6 +60,177 @@
+            raise_if_invalid_filename(event_id_str)
+        except ValueError:
+            event_id_str = ""
+        time_str = str(int(event.timestamp.timestamp()))
+        event_type_str = EVENT_NAME_MAP.get(event.event_type, "event")
+        suffix = "jpg" if event.event_image_type == EventImageType.IMAGE else "mp4"
+        return f"{device_id_str}/{time_str}-{event_id_str}-{event_type_str}.{suffix}"
+
+    def get_media_filename(self, media_key: str) -> str:
+        """Return the filename in storage for a media key."""
+        return f"{self._media_path}/{media_key}"
+
+    async def async_load_media(self, media_key: str) -> bytes | None:
+        """Load media content."""
+        filename = self.get_media_filename(media_key)
+
+        def load_media(filename: str) -> bytes | None:
+            if not os.path.exists(filename):
+                return None
+            _LOGGER.debug("Reading event media from disk store: %s", filename)
+            with open(filename, "rb") as media:
+                return media.read()
+
+        try:
+            return await self._hass.async_add_executor_job(load_media, filename)
+        except OSError as err:
+            _LOGGER.error("Unable to read media file: %s %s", filename, err)
+            return None
+
+    async def async_save_media(self, media_key: str, content: bytes) -> None:
+        """Write media content."""
+        filename = self.get_media_filename(media_key)
+
+        def save_media(filename: str, content: bytes) -> None:
+            os.makedirs(os.path.dirname(filename), exist_ok=True)
+            if os.path.exists(filename):
+                _LOGGER.debug(
+                    "Event media already exists, not overwriting: %s", filename
+                )
+                return
+            _LOGGER.debug("Saving event media to disk store: %s", filename)
+            with open(filename, "wb") as media:
+                media.write(content)
+
+        try:
+            await self._hass.async_add_executor_job(save_media, filename, content)
+        except OSError as err:
+            _LOGGER.error("Unable to write media file: %s %s", filename, err)
+
+    async def async_remove_media(self, media_key: str) -> None:
+        """Remove media content."""
+        filename = self.get_media_filename(media_key)
+
+        def remove_media(filename: str) -> None:
+            if not os.path.exists(filename):
+                return None
+            _LOGGER.debug("Removing event media from disk store: %s", filename)
+            os.remove(filename)
+
+        try:
+            await self._hass.async_add_executor_job(remove_media, filename)
+        except OSError as err:
+            _LOGGER.error("Unable to remove media file: %s %s", filename, err)
+
+    async def _get_devices(self) -> Mapping[str, str]:
+        """Return a mapping of nest device id to home assistant device id."""
+        device_registry = await self._hass.helpers.device_registry.async_get_registry()
```

Comment:
There are callback equivalent now: `async_get`. Also don't use the `hass.helpers` helper as it messes with typing. Better just import it.

```
from homeassistant.helpers import device_registry

dr = device_registry.async_get(self._hass)
```

Reply:
👍🏼